pkgs/tools/networking/ddclient/default.nix

```diff
@@ -20,7 +20,8 @@ perlPackages.buildPerlPackage rec {
     substituteInPlace ddclient \
       --replace 'in the output of ifconfig' 'in the output of ip addr show' \
       --replace 'ifconfig -a'               '${iproute}/sbin/ip addr show' \
-      --replace 'ifconfig $arg'             '${iproute}/sbin/ip addr show $arg'
+      --replace 'ifconfig $arg'             '${iproute}/sbin/ip addr show $arg' \
+      --replace '/usr/bin/perl'		    '${perl}/bin/perl'
```

Comment:
I don't see the text `/usr/bin/perl` in `ddclient` when I install it.

Reply:
@aanderse Are you sure you are getting it from the unstable? 18.09 still have the older 3.8.3 version, which works fine.

```
# nix-env -iA unstable.ddclient
installing 'perl5.28.2-ddclient-3.9.0'
these paths will be fetched (0.04 MiB download, 0.21 MiB unpacked):
  /nix/store/x95bjdmipq9b7z29azlvil7ma2xbsk5f-perl5.28.2-ddclient-3.9.0
copying path '/nix/store/x95bjdmipq9b7z29azlvil7ma2xbsk5f-perl5.28.2-ddclient-3.9.0' from 'https://cache.nixos.org'...
building '/nix/store/4k31zy1lh50lvyldmjhs7b926f333g6a-user-environment.drv'...
created 355 symlinks in user environment

# grep -c '/usr/bin/perl' $(which ddclient)
1
# 
```

Reply:
Sorry I feel silly now. I went to the linked website which linked github for source code... apparently that has the old version and this is why I got tripped up. Ignore me!

Reply:
Don't feel bad, their website is a mess. I spend few minutes looking for the latest release myself.

Reply:
Back to being confused :confused: 
Looking at the git history I see the most recent commit being 6ccc6c7cd11534c6a5f3e83e6c31b6cbdc07114f...
```
~/nixpkgs> git checkout 6ccc6c7cd11534c6a5f3e83e6c31b6cbdc07114f
~/nixpkgs> nix-build -A ddclient
/nix/store/m1lqbvzh4yqnb46hwrjqxnfxybpr433m-perl5.28.0-ddclient-3.9.0
~/nixpkgs> head -1 /nix/store/m1lqbvzh4yqnb46hwrjqxnfxybpr433m-perl5.28.0-ddclient-3.9.0/bin/ddclient 
#!/nix/store/isirn78igzhrpi0rsvjml26rm1hsbwqp-perl-5.28.0/bin/perl -w -I/nix/store/isirn78igzhrpi0rsvjml26rm1hsbwqp-perl-5.28.0/lib/perl5/site_perl -I/nix/store/pz567ql199j8slid15g9mqhclkwxcmkj-perl5.28.0-IO-Socket-SSL-2.060/lib/perl5/site_perl -I/nix/store/d4m0jxhp63a8s9ac4yjk8sl0nfpfdpyr-perl5.28.0-Mozilla-CA-20180117/lib/perl5/site_perl -I/nix/store/b0h9hrfsvzpc4yizpqd55rgbhbjgzcwk-perl5.28.0-Net-SSLeay-1.85/lib/perl5/site_perl -I/nix/store/9cxlyb1p8f2nnly55m3w6qapxcz76v83-perl5.28.0-Digest-SHA1-2.13/lib/perl5/site_perl -I/nix/store/r9g6mszrfri0yq7a85b9i4v4s9kqsf67-perl5.28.0-Data-Validate-IP-0.27/lib/perl5/site_perl -I/nix/store/am9x0hjv4pgwjwc8zqc2k8rjdn38as29-perl5.28.0-NetAddr-IP-4.079/lib/perl5/site_perl -I/nix/store/adqnv2qki4fgkhjf3s8xr8kb68gxyagr-perl5.28.0-JSON-PP-2.97001/lib/perl5/site_perl -I/nix/store/isirn78igzhrpi0rsvjml26rm1hsbwqp-perl-5.28.0/lib/perl5/site_perl -I/nix/store/pz567ql199j8slid15g9mqhclkwxcmkj-perl5.28.0-IO-Socket-SSL-2.060/lib/perl5/site_perl -I/nix/store/d4m0jxhp63a8s9ac4yjk8sl0nfpfdpyr-perl5.28.0-Mozilla-CA-20180117/lib/perl5/site_perl -I/nix/store/b0h9hrfsvzpc4yizpqd55rgbhbjgzcwk-perl5.28.0-Net-SSLeay-1.85/lib/perl5/site_perl -I/nix/store/9cxlyb1p8f2nnly55m3w6qapxcz76v83-perl5.28.0-Digest-SHA1-2.13/lib/perl5/site_perl -I/nix/store/r9g6mszrfri0yq7a85b9i4v4s9kqsf67-perl5.28.0-Data-Validate-IP-0.27/lib/perl5/site_perl -I/nix/store/am9x0hjv4pgwjwc8zqc2k8rjdn38as29-perl5.28.0-NetAddr-IP-4.079/lib/perl5/site_perl -I/nix/store/adqnv2qki4fgkhjf3s8xr8kb68gxyagr-perl5.28.0-JSON-PP-2.97001/lib/perl5/site_perl -I/nix/store/isirn78igzhrpi0rsvjml26rm1hsbwqp-perl-5.28.0/lib/perl5/site_perl -I/nix/store/pz567ql199j8slid15g9mqhclkwxcmkj-perl5.28.0-IO-Socket-SSL-2.060/lib/perl5/site_perl -I/nix/store/d4m0jxhp63a8s9ac4yjk8sl0nfpfdpyr-perl5.28.0-Mozilla-CA-20180117/lib/perl5/site_perl -I/nix/store/b0h9hrfsvzpc4yizpqd55rgbhbjgzcwk-perl5.28.0-Net-SSLeay-1.85/lib/perl5/site_perl -I/nix/store/9cxlyb1p8f2nnly55m3w6qapxcz76v83-perl5.28.0-Digest-SHA1-2.13/lib/perl5/site_perl -I/nix/store/r9g6mszrfri0yq7a85b9i4v4s9kqsf67-perl5.28.0-Data-Validate-IP-0.27/lib/perl5/site_perl -I/nix/store/am9x0hjv4pgwjwc8zqc2k8rjdn38as29-perl5.28.0-NetAddr-IP-4.079/lib/perl5/site_perl -I/nix/store/adqnv2qki4fgkhjf3s8xr8kb68gxyagr-perl5.28.0-JSON-PP-2.97001/lib/perl5/site_perl -I/nix/store/m1lqbvzh4yqnb46hwrjqxnfxybpr433m-perl5.28.0-ddclient-3.9.0/lib/perl5/site_perl
```
So `patchShebangs` is doing its job as of that commit, which is the most recent. Since nothing has changed in `ddclient` then what has changed causing `patchShebangs` in `ddclient` to no longer work, and is the problem more widespread than `ddclient`?

Reply:
@aanderse this is related to the issue #55786 (the shebang is too long to get picked up in the hydra build https://hydra.nixos.org/build/93701252), until we get that shebang script fixed, we need to replace it manually.

Reply:
Right I had forgot about that issue. Thanks again!